runtime/ui/app.go

```diff
@@ -1,149 +1,159 @@
 
-		controller.views.Status.AddHelpKeys(globalHelpKeys...)
+		// Implementation notes: should we factor out this setup??
+		// Probably yes, but difficult to make this both easy to setup & mutable
+		leftVisibleGrid := components.NewVisibleFlex()
+		leftVisibleGrid.SetDirection(tview.FlexRow)
+		rightVisibleGrid := components.NewVisibleFlex()
+		rightVisibleGrid.SetDirection(tview.FlexRow)
+		totalVisibleGrid := components.NewVisibleFlex()
 
-		// perform the first update and render now that all resources have been loaded
-		err = controller.UpdateAndRender()
-		if err != nil {
-			return
+		leftVisibleGrid.AddItem(layersBox, 0, 3, true).
+			AddItem(layerDetailsBox, 0, 1, false).
+			AddItem(imageDetailsBox, 0, 1, false).
+			SetConsumers(layerDetailsBox, layersBox).
+			SetConsumers(imageDetailsBox, layersBox)
+
+		rightVisibleGrid.AddItem(fileTreeBox, 0, 1, false).
+			AddItem(filterView, 1, 0, false).
+			SetConsumers(filterView, fileTreeBox)
+
+		totalVisibleGrid.AddItem(leftVisibleGrid, 0, 1, true).
+			AddItem(rightVisibleGrid, 0, 1, false)
+
+		appSingleton = &diveApp{
+			app:      app,
+			fileTree: fileTreeBox,
+			layers:   layersBox,
 		}
 
+		switchFocus := func(event *tcell.EventKey) *tcell.EventKey {
+			var result *tcell.EventKey = nil
+			switch event.Key() {
+			case tcell.KeyTAB:
+				if appSingleton.app.GetFocus() == appSingleton.layers {
+					appSingleton.app.SetFocus(appSingleton.fileTree)
+				} else {
+					appSingleton.app.SetFocus(appSingleton.layers)
+				}
+			case tcell.KeyCtrlF:
+				if filterView.HasFocus() {
+					filterView.Blur()
+					appSingleton.app.SetFocus(fileTreeBox)
+				} else {
+					appSingleton.app.SetFocus(filterView)
+				}
+
+			default:
+				result = event
+			}
+			return result
+		}
+
+		totalVisibleGrid.SetInputCapture(switchFocus)
+
+		app.SetRoot(totalVisibleGrid, true)
+		app.SetFocus(totalVisibleGrid)
 	})
 
 	return appSingleton, err
 }
 
-// var profileObj = profile.Start(profile.MemProfile, profile.ProfilePath("."), profile.NoShutdownHook)
-// var onExit func()
-
-// debugPrint writes the given string to the debug pane (if the debug pane is enabled)
-// func debugPrint(s string) {
-// 	if controllers.Tree != nil && controllers.Tree.gui != nil {
-// 		v, _ := controllers.Tree.gui.View("debug")
-// 		if v != nil {
-// 			if len(v.BufferLines()) > 20 {
-// 				v.Clear()
-// 			}
-// 			_, _ = fmt.Fprintln(v, s)
-// 		}
-// 	}
-// }
-
-// quit is the gocui callback invoked when the user hits Ctrl+C
-func (a *app) quit() error {
-
-	// profileObj.Stop()
-	// onExit()
-
-	return gocui.ErrQuit
-}
-
 // Run is the UI entrypoint.
-func Run(analysis *image.AnalysisResult, treeStack filetree.Comparer) error {
-	var err error
-
-	g, err := gocui.NewGui(gocui.OutputNormal)
+func Run(analysis *image.AnalysisResult, treeStack filetree.Comparer, isCNB bool) error {
+	cfg := zap.NewDevelopmentConfig()
+	cfg.OutputPaths = []string{"/tmp/dive/debug.out"}
```

Comment:
```suggestion
	cfg := zap.NewDevelopmentConfig()
	os.MkdirAll("/tmp/dive", os.ModePerm)
	cfg.OutputPaths = []string{"/tmp/dive/debug.out"}
```

I added it to correct the error below.

> panic: couldn't open sink ... no such file or directory

Reply:
:+1:  much appreciated